src/manager.cpp

```diff
@@ -185,10 +186,12 @@ sdbusplus::async::task<>
 
         while (!_ctx.stop_requested())
         {
-            if (co_await dataWatcher.onDataChange())
-            {
-                co_await syncData(dataSyncCfg);
-            }
+            auto dataOperations = co_await dataWatcher.onDataChange();
```

Comment:
it could be written as

```
if (auto dataOperations = co_await dataWatcher.onDataChange(); !dataOperations.empty())
{
std::ranges::for_each(dataOperations, [&dataSyncCfg]([[maybe_unused]] const auto& dataOp)
                -> sdbusplus::async::task<> { co_await syncData(dataSyncCfg); });
}
```

Why are we returning the operation but not using it? I suppose it would be useful to notify which specific file or directory has changed. This way, only the affected file or directory could be reloaded instead of reloading the entire configured directory. Additionally, it could help adjust the path for rsync to properly handle deletions, right?